lib/index.js

```diff
@@ -121,75 +164,97 @@ export class Kadira {
       data: stream,
       headers: {
         ...this._headers,
-        'content-type': 'application/octet-stream'
+        'content-type': ContentType.STREAM
       },
     };
 
     logger(`send stream to ${url}`);
     return this._send(url, params);
   }
 
+  _handleJobEvent(job) {
+    if (this._jobQueue.find(j => j._id === job._id)) {
+      return;
+    }
+
+    this._jobQueue.push(job);
+
+    this.emit(CoreEvent.JOB_ADDED, job);
+  }
+
+  _handleMessage(message) {
+    try {
+      if (!message.data) {
+        return;
+      }
+
+      if ([ 'ping', 'pong' ].includes(message.data.toString())) {
+        return;
+      }
+
+      const { event, data } = JSON.parse(message.data);
+
+      switch (event) {
+        case EngineEvent.JOB_CREATED:
+          this._handleJobEvent(data);
+      }
+    } catch (error) {
+      console.log(
+        'Monti APM: Failed to parse message',
+        JSON.stringify(message.data)
+      );
+      console.log(error.stack);
+    }
+  }
+
+  _initWebSocket() {
+    if (!this.featureSupported(Feature.WEBSOCKETS)) {
+      return;
+    }
+
+    if (this._ws) {
+      return;
+    }
+
+    const { disconnect } = persistentConnectWebSocket(
+      this._options.endpoint,
+      this._websocketHeaders,
+      this._handleMessage.bind(this)
+    );
+
+    this.removeAllListeners(CoreEvent.DISCONNECT);
+
+    this.on(CoreEvent.DISCONNECT, disconnect);
```

Comment:
It might be nice to handle disconnecting in the `Kadira#disconnect` method.

Reply:
The problem is that we don't have a direct reference for the WS instance, I did change it so it listens to the disconnect event coming from the the disconnect method.